src/Bridges/Constraint/bridges/inequality_to_complements.jl

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    InequalityToComplementsBridge{T,F,S,G} <: Bridges.Constraint.AbstractBridge
+
+`InequalityToComplementsBridge` implements the following reformulations:
+
+  * ``f(x) \\ge b`` into ``f(x) - b \\perp y \\ge 0``
+  * ``f(x) \\le b`` into ``f(x) - b \\perp y \\le 0``
+  * ``f(x) = b`` into ``f(x) - b \\perp y``
+
+## Source node
+
+`InequalityToComplementsBridge` supports:
+
+  * `F` in [`MOI.GreaterThan{T}`](@ref)
+  * `F` in [`MOI.LessThan{T}`](@ref)
+  * `F` in [`MOI.EqualTo`](@ref)
+
+## Target nodes
+
+`InequalityToComplementsBridge` creates:
+
+  * [`MOI.VariableIndex`](@ref) in [`MOI.LessThan{T}`](@ref)
+  * [`MOI.VariableIndex`](@ref) in [`MOI.GreaterThan{T}`](@ref)
+  * `G` in [`MOI.Complements`](@ref)
+"""
+mutable struct InequalityToComplementsBridge{
+    T,
+    F<:MOI.AbstractScalarFunction,
+    S<:Union{MOI.GreaterThan{T},MOI.LessThan{T},MOI.EqualTo{T}},
+    G<:MOI.AbstractVectorFunction,
+} <: AbstractBridge
+    y::MOI.VariableIndex
+    set::S
+    ci::MOI.ConstraintIndex{G,MOI.Complements}
+end
+
+const InequalityToComplements{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{InequalityToComplementsBridge{T},OT}
+
+function _add_y_variable(model, ::MOI.GreaterThan{T}) where {T}
+    return MOI.add_constrained_variable(model, MOI.GreaterThan(zero(T)))[1]
+end
+
+function _add_y_variable(model, ::MOI.LessThan{T}) where {T}
+    return MOI.add_constrained_variable(model, MOI.LessThan(zero(T)))[1]
+end
+
+_add_y_variable(model, ::MOI.EqualTo) = MOI.add_variable(model)
+
+function bridge_constraint(
+    ::Type{InequalityToComplementsBridge{T,F,S,G}},
+    model::MOI.ModelLike,
+    f::F,
+    set::S,
+) where {T,F,S<:Union{MOI.GreaterThan{T},MOI.LessThan{T},MOI.EqualTo{T}},G}
+    y = _add_y_variable(model, set)
+    f_set = MOI.Utilities.operate(-, T, f, MOI.constant(set))
+    g = MOI.Utilities.operate(vcat, T, f_set, y)
```

Comment:
```suggestion
    g = MOI.Utilities.operate(vcat, T, y, f_set)
```

Reply:
I find it more consistent with indicator where the variable is first

Reply:
No, the order is important, and it is not interchangeable. (We chose this order because people always write F(x) perp x, not x perp F(x).

https://github.com/jump-dev/MathOptInterface.jl/blob/f6f8a09cedb3d8b2e10e18a4ff6e2440ec6954cb/src/sets.jl#L1962-L2047